app/services/curation_concerns/curation_concern.rb

```diff
@@ -1,9 +1,20 @@
 module CurationConcerns
   module CurationConcern
-    def self.actor(curation_concern, *args)
+    # Returns the top-level actor on the stack
+    def self.actor(curation_concern, current_user, attributes)
+      AddToCollectionActor.new(curation_concern, current_user, attributes,
```

Comment:
^ Right, this is the concerning bit for me. This might be my functional programming side talking, but actors should probably be units with a common API, which another pattern can take advantage of to move between them.